motleycrew/agents/parent.py

```diff
@@ -188,6 +195,26 @@ def materialize(self):
         else:
             self._agent = self.agent_factory(tools=self.tools)
 
+    def prepare_for_invocation(self, input: dict) -> str:
+        """Prepares the agent for invocation by materializing it and composing the prompt.
+
+        Should be called in the beginning of the agent's invoke method.
+
+        Args:
+            input (dict): the input to the agent
+
+        Returns:
+            str: the composed prompt
+        """
+        self.materialize()
+
+        if isinstance(self.output_handler, MotleyOutputHandler):
+            self.output_handler.agent = self
+            self.output_handler.last_agent_input = input
```

Comment:
Maybe rename to agent_input?

Reply:
done

Reply:
the idea was that the agent could be called multiple times esp. if with history, but it's kind of obvious that the output handler is validating the output to the last input :)